token/js/src/extensions/tlvData.ts

```diff
@@ -0,0 +1,95 @@
+export type TLVNumberSize = 1 | 2 | 4 | 8;
+
+function readTLVNumberSize<T>(
+    buffer: Buffer,
+    size: TLVNumberSize,
+    offset: number,
+    constructor: (x: number | bigint) => T
+): T {
+    switch (size) {
+        case 1:
+            return constructor(buffer.readUInt8(offset));
+        case 2:
+            return constructor(buffer.readUInt16LE(offset));
+        case 4:
+            return constructor(buffer.readUInt32LE(offset));
+        case 8:
+            return constructor(buffer.readBigUInt64LE(offset));
+    }
+}
```

Comment:
This is a neat function here, but our TLV library is using `u32` for length and Token2022 is using `u16`, so we can probably do away with handling `u64` and eliminate the need for `bigint`. What do you think?

https://github.com/solana-labs/solana-program-library/blob/f35dc5fdb6fc01d054afc51a6ea5a10307a48e3b/token/program-2022/src/extension/mod.rs#L81

https://github.com/solana-labs/solana-program-library/blob/f35dc5fdb6fc01d054afc51a6ea5a10307a48e3b/libraries/type-length-value/src/length.rs#L11

Reply:
But the discriminator could still by 8 bytes, right? This function is currently used for both parsing the discriminator and the length

Reply:
Yep that makes sense. This was only added because you can then supply a BigInt or number in the `entry` function as well as a Buffer.